### PR TITLE
Add stack configuration docs.

### PR DESCRIPTION
**Pull request by tjholm**
Add configuration docs for stacks to our existing docs, these are currently under overview but there is probably a better place to put them.

Do we also want to include documentation for individual providers as well?

And should we add placeholder configuration to our templates?

### CODE REVIEW

File: docs/config/stack.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    default:
+        # Set functions to have 512 MB of memory
+        memory: 512
+        # Set functions to have a timeout of 30 seconds
+        timeout: 30
+    # Set the memory and timeout for a specific function
+    functions/hello-world.ts:
+        memory: 1024
+        timeout: 15
+```
```
Comment (jyecusch) [inline docs/config/stack.md:21]: It's probably worth noting this config goes in the `nitric.yaml` file (and/or show that in the example).

Also, if wildcards work in the function names e.g. `functions/hello-*.ts` can we document that too? If they're not supported, we should probably create a new issue to consider adding that support.

**Review by jyecusch — CHANGES_REQUESTED** [review]

File: docs/config/stack.md (hunk 1 above)
Comment (tjholm) [reply docs/config/stack.md:21]: cool, will add that note. At the moment this configuration actually goes in the nitric-*.yaml stack file, but we could extend this to the base file and add inheritance for the stack files. The assumption being that not all clouds would have the same infrastructure requirements.

File: docs/config/stack.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+## Stack Configuration
+
+
+### Function Config
+
+Functions can be configured with memory and timeout requirements per function, a default can also be set for all functions in a stack that do not have their own configuration.
```
Comment (raksiv) [inline docs/config/stack.md:6]: It would probably be worth specifying what the default is when this configuration is not set

Comment (tjholm) [reply docs/config/stack.md:5]: Have added a sub section on default values.

**Review by raksiv — APPROVED** [review]

**Review by asalkeld — APPROVED** [review]

**Pull request closed without merging**